File: Python/Algorithms/1662.py
from itertools import chain, zip_longest


class Solution:
    def arrayStringsAreEqual(self, word1, word2):
        chain_one = chain.from_iterable(word1)
        chain_two = chain.from_iterable(word2)

        for ca, cb in zip_longest(chain_one, chain_two):
            if ca != cb:
                return False
        return True

File: Rust/Algorithms/1662.rs
// This is damn pretty. 
impl Solution {
    pub fn array_strings_are_equal(word1: Vec<String>, word2: Vec<String>) -> bool {
        word1
            .iter()
            .flat_map(|s| s.chars())
            .eq(word2.iter().flat_map(|s| s.chars()))
    }
}
